plugin/src/main/java/io/fabric8/maven/plugin/HelmMojo.java

```diff
@@ -0,0 +1,248 @@
+                    FilenameFilter filter = new FilenameFilter() {
+                        @Override
+                        public boolean accept(File dir, String name) {
+                            return name.toLowerCase(Locale.ENGLISH).startsWith("readme.");
+                        }
+                    };
+                    copyTextFile(basedir, filter, new File(outputDir, outputReadMeFileName));
+                } catch (IOException e) {
+                    throw new MojoExecutionException("Failed to save " + outputReadMeFileName + ". Reason: " + e, e);
+                }
+                String outputLicenseFileName = "LICENSE";
+                try {
+                    FilenameFilter filter = new FilenameFilter() {
+                        @Override
+                        public boolean accept(File dir, String name) {
+                            String lower = name.toLowerCase(Locale.ENGLISH);
+                            return lower.equals("license") || lower.startsWith("license.");
+                        }
+                    };
+                    copyTextFile(basedir, filter, new File(outputDir, outputLicenseFileName));
+                } catch (IOException e) {
+                    throw new MojoExecutionException("Failed to save " + outputLicenseFileName + ". Reason: " + e, e);
+                }
+            }
+        }
+
+        getLog().info("Generated Helm Chart " + chartName + " at " + outputDir);
+
+        // now lets create the tarball
+        if (basedir != null) {
+            createTarGzip(outputDir, new File(basedir, "target/" + chartName + "-" + project.getVersion() + "-" + classifier + ".tar.gz"), classifier);
+        }
+
+    }
+
+    protected void createTarGzip(File sourceDir, File outputFile, String classifier) throws MojoExecutionException {
+        try {
+            archiver.setCompression(TarArchiver.TarCompressionMethod.gzip);
+            archiver.setLongfile(TarLongFileMode.posix);
+            archiver.addDirectory(sourceDir);
+            archiver.setDestFile(outputFile);
+            archiver.createArchive();
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to create helm archive " + outputFile + ". " + e, e);
+        }
+        projectHelper.attachArtifact(project, "tar.gz", classifier, outputFile);
+    }
+
+    private boolean containsYamlFiles(File sourceDir) {
+        File[] files = sourceDir.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                String lower = file.getName().toLowerCase();
+                if (lower.endsWith(".yaml") || lower.endsWith(".yml")) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    protected static File copyTextFile(File sourceDir, FilenameFilter filter, File outFile) throws IOException {
+        File[] files = sourceDir.listFiles(filter);
+        if (files != null && files.length == 1) {
```

Comment:
what if there are more than one README or LICENSE ? Shouldnt we pick one ? At least we should add a warning. The return value is ignored,  btw.